org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/QuickAssistProcessor.java

```diff
@@ -2046,182 +2045,28 @@ private static boolean getSplitVariableProposals(IInvocationContext context, AST
-		ASTRewrite rewrite= ASTRewrite.create(ast);
-
-		String label= CorrectionMessages.QuickAssistProcessor_splitdeclaration_description;
-		Image image= JavaPluginImages.get(JavaPluginImages.IMG_CORRECTION_LOCAL);
-		ASTRewriteCorrectionProposal proposal= new ASTRewriteCorrectionProposal(label, context.getCompilationUnit(), rewrite, IProposalRelevance.SPLIT_VARIABLE_DECLARATION, image);
-		boolean commandConflict= false;
-		for (ICommandAccess completionProposal : resultingCollections) {
-			if (completionProposal instanceof ChangeCorrectionProposal) {
-				if (SPLIT_JOIN_VARIABLE_DECLARATION_ID.equals(((ChangeCorrectionProposal) completionProposal).getCommandId())) {
-					commandConflict= true;
-				}
-			}
-		}
-		if (!commandConflict) {
-			proposal.setCommandId(SPLIT_JOIN_VARIABLE_DECLARATION_ID);
-		}
-
-		// for multiple declarations; all must be moved outside, leave none behind
-		if (statement instanceof ForStatement) {
-			IBuffer buffer= context.getCompilationUnit().getBuffer();
-			ForStatement forStatement= (ForStatement) statement;
-			VariableDeclarationExpression oldVarDecl= (VariableDeclarationExpression) fragParent;
-			Type type= oldVarDecl.getType();
-			ITypeBinding tBinding= type.resolveBinding();
-			List<VariableDeclarationFragment> oldFragments= oldVarDecl.fragments();
-			CompilationUnit cup= (CompilationUnit) fragment.getRoot();
-			ListRewrite forListRewrite= rewrite.getListRewrite(forStatement, ForStatement.INITIALIZERS_PROPERTY);
-			// create the new initializers
-			for (VariableDeclarationFragment oldFragment : oldFragments) {
-				int extendedStartPositionFragment= cup.getExtendedStartPosition(oldFragment);
-				int extendedLengthFragment= cup.getExtendedLength(oldFragment);
-				StringBuilder codeFragment= new StringBuilder(buffer.getText(extendedStartPositionFragment, extendedLengthFragment));
-				if (oldFragment.getInitializer() == null) {
-					ITypeBinding typeBinding= type.resolveBinding();
-					if ("Z".equals(typeBinding.getBinaryName())) { //$NON-NLS-1$
-						codeFragment.append(" = false"); //$NON-NLS-1$
-					} else if (type.isPrimitiveType()) {
-						codeFragment.append(" = 0"); //$NON-NLS-1$
-					} else {
-						codeFragment.append(" = null"); //$NON-NLS-1$
+		VariableDeclarationFixCore fix= VariableDeclarationFixCore.createSplitVariableFix(context.getASTRoot(), node);
+		if (fix != null) {
+			Image image= JavaPluginImages.get(JavaPluginImages.IMG_CORRECTION_LOCAL);
+			Map<String, String> options= new HashMap<>();
+			options.put(CleanUpConstants.CONTROL_STATEMENTS_CONVERT_FOR_LOOP_TO_ENHANCED, CleanUpOptions.TRUE);
+			ICleanUp cleanUp= new ConvertLoopCleanUp(options);
+			FixCorrectionProposal proposal= new FixCorrectionProposal(fix, cleanUp, IProposalRelevance.CONVERT_FOR_LOOP_TO_ENHANCED, image, context);
+			boolean commandConflict= false;
+			for (ICommandAccess completionProposal : resultingCollections) {
+				if (completionProposal instanceof FixCorrectionProposal) {
```

Comment:
This check is to find if there is a Join variable proposal for the same code.  This cannot be a FixCorrectionProposal but is a ChangeCorrectionProposal so you should not change this.  In addition, this check is done before forming the Split fix so you should probably do the same to avoid wasting computation.